### ISSUE TITLE
Unable to connect to the MCP server

### ISSUE THREAD

**Issue by arimariojesus** [modelcontextprotocol/inspector#501]
**Describe the bug**
A Connection Error is triggered when I try to connect to my local MCP server using STDIO transport.

**To Reproduce**
1. Create a custom implementation for STDIO using node
2. Build your code
3. Run inspector using `npx @modelcontextprotocol/inspector node {your-file}`
4. In the inspector GUI click on "Connect"

**Expected behavior**

**Logs**
Stdio transport: command=/Users/{user}/.asdf/installs/nodejs/23.11.0/bin/node, args=./dist/index.js
Spawned stdio transport
Connected MCP client to backing server transport
Created client/server transports
Set up MCP proxy
Received message for sessionId e8fde9c9-ebc1-4bd1-aa4f-1b9b805915f4

**Additional context**
Code of the custom implementation:
```ts
import * as readline from 'node:readline';
import { stdin, stdout } from 'node:process';

const serverInfo = {
  name: 'Example MCP Server',
  version: '1.0.0',
};

const rl = readline.createInterface({
  input: stdin,
  output: stdout,
});

function sendResponse(id: number, result: object) {
  const response = {
    jsonrpc: '2.0',
    result,
    id,
  };
  console.log(JSON.stringify(response));
}

(async function main() {
  for await (const line of rl) {
    try {
      const json = JSON.parse(line);
      if (json.jsonrpc === '2.0') {
        if (json.method === 'initialize') {
          sendResponse(json.id, {
            protocolVersion: '"2025-03-26',
            capabilities: {
              tools: { listChanged: true },
            },
            serverInfo,
          });
        }
      }
    } catch (error) {
      console.error(error);
    }
  }
})();
```

Error in the inspector:
![Image](https://github.com/user-attachments/assets/65c8e80e-8965-4ff2-9e7b-02a5ac17aa9f)

**Comment by arimariojesus** [comment]
I realized I forgot an unwanted double quote in `protocolVersion` value. So when I removed it, the server worked correctly.

Instead of:
```
protocolVersion: '"2025-03-26'
```

I just pass:
```
protocolVersion: '2025-03-26'
```
